controllers/llamastackdistribution_controller.go

```diff
@@ -393,93 +400,154 @@ func (r *LlamaStackDistributionReconciler) getProviderInfo(ctx context.Context,
+	deployment := &appsv1.Deployment{}
+	deploymentErr := r.Get(ctx, types.NamespacedName{Name: instance.Name, Namespace: instance.Namespace}, deployment)
+	if deploymentErr != nil && !k8serrors.IsNotFound(deploymentErr) {
+		return false, fmt.Errorf("failed to fetch deployment for status: %w", deploymentErr)
+	}
+
+	deploymentReady := false
+
+	switch {
+	case deploymentErr != nil: // This case covers when the deployment is not found
+		instance.Status.Phase = llamav1alpha1.LlamaStackDistributionPhasePending
+		SetDeploymentReadyCondition(&instance.Status, false, MessageDeploymentPending)
+	case deployment.Status.ReadyReplicas == 0:
+		instance.Status.Phase = llamav1alpha1.LlamaStackDistributionPhaseInitializing
+		SetDeploymentReadyCondition(&instance.Status, false, MessageDeploymentPending)
+	case deployment.Status.ReadyReplicas < instance.Spec.Replicas:
+		instance.Status.Phase = llamav1alpha1.LlamaStackDistributionPhaseInitializing
+		deploymentMessage := fmt.Sprintf("Deployment is scaling: %d/%d replicas ready", deployment.Status.ReadyReplicas, instance.Spec.Replicas)
+		SetDeploymentReadyCondition(&instance.Status, false, deploymentMessage)
+	case deployment.Status.ReadyReplicas > instance.Spec.Replicas:
+		instance.Status.Phase = llamav1alpha1.LlamaStackDistributionPhaseInitializing
+		deploymentMessage := fmt.Sprintf("Deployment is scaling down: %d/%d replicas ready", deployment.Status.ReadyReplicas, instance.Spec.Replicas)
+		SetDeploymentReadyCondition(&instance.Status, false, deploymentMessage)
+	default:
+		instance.Status.Phase = llamav1alpha1.LlamaStackDistributionPhaseReady
+		deploymentReady = true
+		SetDeploymentReadyCondition(&instance.Status, true, MessageDeploymentReady)
+		if instance.Status.Version.LlamaStackVersion == "" {
+			instance.Status.Version.LlamaStackVersion = os.Getenv("LLAMA_STACK_VERSION")
+		}
+	}
+	instance.Status.AvailableReplicas = deployment.Status.ReadyReplicas
+	return deploymentReady, nil
+}
+
+func (r *LlamaStackDistributionReconciler) updateStorageStatus(ctx context.Context, instance *llamav1alpha1.LlamaStackDistribution) {
+	if instance.Spec.Server.Storage == nil {
+		return
+	}
+	pvc := &corev1.PersistentVolumeClaim{}
+	err := r.Get(ctx, types.NamespacedName{Name: instance.Name + "-pvc", Namespace: instance.Namespace}, pvc)
+	if err != nil {
+		SetStorageReadyCondition(&instance.Status, false, fmt.Sprintf("Failed to get PVC: %v", err))
+		return
+	}
+
+	ready := pvc.Status.Phase == corev1.ClaimBound
+	var message string
+	if ready {
+		message = MessageStorageReady
+	} else {
+		message = fmt.Sprintf("PVC is not bound: %s", pvc.Status.Phase)
+	}
+	SetStorageReadyCondition(&instance.Status, ready, message)
+}
+
+func (r *LlamaStackDistributionReconciler) updateServiceStatus(ctx context.Context, instance *llamav1alpha1.LlamaStackDistribution) {
+	log := log.FromContext(ctx).WithValues("namespace", instance.Namespace, "name", instance.Name)
```

Comment:
```suggestion
	logger := log.FromContext(ctx)
```

We tend to always call it logger.